### PR TITLE
Zip Deploy support for web apps

### PR DESCRIPTION
**Pull request by SumiranAgg**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/task.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "version": {
         "Major": 4,
         "Minor": 1,
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/task.json]: nit: increase minor version.

File: Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+        if(runFromZip){
+            var customApplicationSetting = ParameterParser.parse(runFromZipAppSetting);
+            await this.appServiceUtility.updateAndMonitorAppSettings(customApplicationSetting);
+        }
+
+        this.zipDeploymentID = await this.kuduServiceUtility.zipDeploy(webPackage, runFromZip, this.taskParams.TakeAppOfflineFlag, 
+            { slotName: this.appService.getSlot() });
+
+        await this.appServiceUtility.updateStartupCommandAndRuntimeStack(this.taskParams.RuntimeStack, this.taskParams.StartupCommand);
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts]: this shouldnt be invoked for Windows app

File: Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            this.deployUsingZipDeploy(webPackage, true);
+        }
+
+        await this.PostDeploymentStep();
+    }
+
+    private async deployUsingZipDeploy(webPackage, runFromZip: boolean) {
+        tl.debug("Initiated deployment via kudu service for webapp package : ");
+
+        if(runFromZip){
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts]: nit: space before {

File: Tasks/AzureRmWebAppDeployment/operations/TaskParameters.ts [hunk 4]
```diff
@@ -8,7 +9,7 @@ export class TaskParametersUtility {
             WebAppKind: tl.getInput('WebAppKind', false),
             DeployToSlotOrASEFlag: tl.getBoolInput('DeployToSlotOrASEFlag', false),
             VirtualApplication: tl.getInput('VirtualApplication', false),
-            Package: tl.getPathInput('Package', true),
+            Package: new Package(tl.getPathInput('Package', true)),
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/operations/TaskParameters.ts]: package is not required for container app deployment. We might want to set null in this case.

File: Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            await this.deployUsingZipDeploy(webPackage, this.taskParams.UseRunFromZip);
+        }
+        else if(this.taskParams.ScriptType) {
+            await this.deployUsingZipDeploy(webPackage, false);
+        }
+        else {
+            await this.deployUsingZipDeploy(webPackage, true);
+        }
+
+        await this.PostDeploymentStep();
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts]: no post deployment script in RunFromZip

Comment (SumiranAgg) [reply Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts]: There are two cases when runFromZip will be used:
- When user explicitly selects from the task, in that case we dont allow user to add post deployment script.
- When we do auto-detect, as you see the logic if the user adds the deployment script we are calling zipdeploy with runFromZip flag value as "false"
Do you see any other scenario that I am missing?

Comment (vincentdass) [reply Tasks/AzureRmWebAppDeployment/deploymentProvider/WindowsWebAppZipDeployProvider.ts]: we might want to fallback  to puer zipdeploy on auto-detect when Post deployment script is present. @kmkumaran , @azooinmyluggage  , can you please share your thoughts?

**Pull request merged**